src/pcapi/utils/cancellation_date.py

```diff
@@ -0,0 +1,24 @@
+from datetime import datetime
+from typing import Optional
+
+from pcapi.core.bookings.api import compute_confirmation_date
+from pcapi.utils.date import format_into_utc_date, match_format
+
+
+def get_cancellation_limit_date(beginning_datetime: str, cancellation_limit_date: Optional[str] = None) -> Optional[str]:
+    if not beginning_datetime:
+        return None
+
+    if not cancellation_limit_date:
+        is_right_format = match_format(beginning_datetime, "%Y-%m-%dT%H:%M:%SZ")
+
```

Comment:
```suggestion
        if match_format(beginning_datetime, DATE_ISO_FORMAT):
            transformed_beginning_datetime = datetime.strptime(beginning_datetime, DATE_ISO_FORMAT)
        else:
            transformed_beginning_datetime = datetime.strptime(beginning_datetime, "%Y-%m-%dT%H:%M:%SZ")
```
?
Ça permettrait d'éviter la duplication de la chaîne de caractères "%Y-%m-%dT%H:%M:%SZ"

Reply:
Je trouve dommage de faire appel à `strptime()`. Est-ce qu'on ne pourrait pas placer l'appel à cette fonction à un endroit où Pydantic a déjà converti (ou rejeté !) la string en datetime ? Ça me semblerait plus propre.

Reply:
Ici la méthode est appelée hors d'usage de Pydantic :/